applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_ramm_arc_length_strategy.hpp

```diff
@@ -534,6 +541,23 @@ class GeoMechanicsRammArcLengthStrategy
         // Save the applied Lambda factor
         mLambda_old = mLambda;
     }
+
+private:
+    static double CalculateReferenceDofsNorm(DofsArrayType& rDofSet)
+    {
+        auto is_free_dof = [](const auto& rDof) { return rDof.IsFree(); };
+        auto free_dofs   = rDofSet | boost::adaptors::filtered(is_free_dof);
+
+        auto free_dof_values = Vector{boost::size(free_dofs)};
+        auto get_dof_value   = [](const auto& rDof) { return rDof.GetSolutionStepValue(); };
+        std::transform(std::begin(free_dofs), std::end(free_dofs), free_dof_values.begin(), get_dof_value);
+
+        return norm_2(free_dof_values);
+    }
+
+    std::vector<ModelPart*> mSubModelPartList; // List of every SubModelPart associated to an external load
+    std::vector<std::string> mVariableNames; // Name of the nodal variable associated to every SubModelPart
```

Comment:
Was this used anywhere in the end?

Reply:
Yes, I've found these data members are being used. See member function `UpdateExternalLoads`, which uses the lists of model part pointers and variable names. Unfortunately, there are no tests that cover this code :-(